lib/rules/arrow-body-style.js

```diff
@@ -99,6 +100,19 @@ module.exports = {
             return sourceCode.getTokenAfter(node);
         }
 
+        /**
+         * Check whether the node is inside of a for loop
+         * @param {ASTNode} node node is inside for loop
+         * @returns {boolean} `true` if the node is inside of a for loop, else `false`
+         */
+        function isInsideForLoopInitializer(node) {
+            if (node.type !== "ForStatement" && node.parent && (node.parent.type !== "BlockStatement" || node.parent.type !== "Program")) {
+                return isInsideForLoopInitializer(node.parent);
+            }
+
+            return node.type === "ForStatement";
+        }
```

Comment:
Looks good, can we just additionally make sure that it's in `ForStatement#init` (since this function now also returns `true` if it's in `test`, `update`, or `body`).

Reply:
done